### DISCUSSION TITLE
vips_text with different horizontal and vertical dpi

### DISCUSSION THREAD

**Discussion by Bytologic** [libvips/libvips#2861] [Ideas]
Hi,

For an application I am working on I'd like to generate text with different horizontal and vertical resolutions. What I am currently doing is giving the highest of the two resolutions to vips_text and then downscale the result. However, performance is a big concern for my application. The change I am proposing is to add another optional argument to vips_text, maybe called "vdpi" in the same vein as resize having "scale" and an optional "vscale". This way I would be able to skip the scaling part and hopefully get better performance. Would something like this be possible?

**Comment by jcupitt** [comment]
Hi @Bytologic 

libvips uses `pango_cairo_font_map_set_resolution()` to set the rendering scale:

https://developer.gimp.org/api/2.0/pango/pango-Cairo-Rendering.html

And I don't see a simple way to give horizontal and vertical resolution separately.

Maybe you could make a custom font with the stretch you need?

**Reply by Bytologic** [reply]
Thank you for your quick response.

Unfortunately these resolutions are selected by a user so creating different fonts for every combination does not work for me.

After some searching I did find `pango_ft2_font_map_set_resolution()`, which could be used to set different resolutions:

https://developer.gimp.org/api/2.0/pango/pango-FreeType-Fonts-and-Rendering.html

I do not know if it's possible to use this or if that would be a major change for libvips.
Would this be a feasable solution to this problem?

**Reply by Bytologic** [reply]
Hi @jcupitt ,

Sorry to bother you again.
Were you able to take a look into this suggestion?

**Reply by jcupitt** [reply]
Hi again, we're just getting libvips 8.13 out of the door. Perhaps next week.
